File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    final int BASE_HOURLY_RATE = 221;
    
    public double productionRatePerHour(int speed) {
	double successRate;
        if (speed < 5) {
	    successRate = 1;
	} else if (speed < 9) {
	    successRate = 0.9;
	} else if (speed == 9) {
	    successRate = 0.8;
	} else {
	    successRate = 0.77;
	}
	return BASE_HOURLY_RATE * successRate * speed;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)Math.floor(productionRatePerHour(speed) / 60.0);
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed) => speed switch 
    {
        0 => 0.0,
        < 5 => 1.0,
        < 9 => 0.9,
        9 => 0.8,
        10 => 0.77,
        _ => 0.0,
    };
    
    public static double ProductionRatePerHour(int speed) => 221.0 * speed * SuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed) => (int)Math.Floor(ProductionRatePerHour(speed) / 60.0);
}
